### DISCUSSION TITLE
Week 08

### DISCUSSION THREAD

**Discussion by JerryNixon** [ColoradoChristianUniversity/CIS-230-SP24#71] [General]
Week 08

**Comment by M10377** [comment]


**Reply by Alex-0-Baxter** [reply]
// num 5

else if (selection == 5) // "Which versions of .NET were released in a leap year?"
    {
        
        var leap = dates.Where(x => (x.Year % 4 == 0) || ( (x.Year % 100 == 0) && (x.Year % 400 == 0) )).First().Name;

        var message = $"{leap} was released on a leap year";
        WriteMessage(message); 
    }

**Comment by 0578327** [comment]
#Jaxson Number 3

else if (selection == 3)
    {
        //"How many versions were released in November?"
        var novVersions = dates.Count(x => x.Month == 11);
        var message = $"There were {novVersions} versions of .NET released in November";
        WriteMessage(message);
    }

**Comment by M10377** [comment]
#7

```
else if (selection == 7)
{
    // "Which months have never been used for a release?"
    var months = dates.Select(x => x.Month).Distinct();
    var unusedMonths = Enumerable.Range(1, 12).Except(months);
    var message = $"The following months have never been used for a release: {string.Join(", ", unusedMonths)}";
    WriteMessage(message);
}
```

**Comment by mjs2002** [comment]
## Statement 9 - Marek S.
```
    else if (selection == 9)
    {
        var releaseDates = dates.Select(date => new DateTime(date.Date.Year, date.Date.Month, date.Date.Day));
        var longestPeriod = releaseDates.Zip(releaseDates.Skip(1), (d1, d2) => (d2 - d1).TotalDays).Max();
        var message = $"The longest period between two versions is {longestPeriod} days.";
        WriteMessage(message);
    }
```

**Comment by IDMunguia** [comment]
Problem 1

var first = dates.First().Name;
var last = dates.Last().Name;
var count = dates.Count();
var message = $"From {first} to {last}, there have been {count} version(s).";
WriteMessage(message);

**Comment by SeFreemanCCU** [comment]
# Number 6
```
else if (selection == 6)
{
    // "What is the average number of days between releases?";
    var daysBetween = new List<int>();

    for (int i = 1; i < dates.Length; i++)
    {
        var previousRelease = new DateTime(dates[i - 1].Year, dates[i - 1].Month, dates[i - 1].Day);
        var currentRelease = new DateTime(dates[i].Year, dates[i].Month, dates[i].Day);
        var daysDifference = (int)(currentRelease - previousRelease).TotalDays;
        daysBetween.Add(daysDifference);
    }

    var averageDays = daysBetween.Average();
    WriteMessage($"The average number of days between releases is {averageDays:F0}.");
}
```

**Comment by jewelsmski** [comment]
### Statement 2
First Version
  int firstVersionYear = versions.Min(v => v.ReleaseDate.Year);
        Console.WriteLine($"Year of the first version of .NET: {firstVersionYear}");

**Comment by syd-codes** [comment]
## Number 4 - Sydney
`  else if (selection == 4)
    {
        //"How many years are there before the first and last versions?"
        var firstVersion = dates
            .OrderBy(x => x.Year)
            .First()
            .Year;
        var lastVersion = dates
            .OrderByDescending(x => x.Year)
            .First()
            .Year;  
        var years = lastVersion - firstVersion;
        var message = $"There are {years} years between the first and last versions of .NET.";
        WriteMessage(message);
    }`

**Comment by masonlink17** [comment]
Mason:

else if (selection == 8)
    // "What day of the week is the most common release day?"

    {
```
        var mostCommonDayOfWeek = dates
        .GroupBy(x => x.Date.DayOfWeek)
        .OrderByDescending(g => g.Count())
        .First()
        .Key;
        

        var message = $"The most common release day of the week is {mostCommonDayOfWeek}.";
        WriteMessage(message);
```
    }

**Comment by GabeWassmer** [comment]
##Gabe Wass - Question 10
    else if (selection == 10)
    {
        // "What is the current version of .NET?"
        var current = dates.Last();
        var message = $"The current version of .NET is {current.Name}.";
        WriteMessage(message);
    }
